tests/simulation/test_simulation_device.cpp

```diff
@@ -38,7 +38,7 @@ TEST_P(LoopbackAllCoresParam, LoopbackSingleTensix){
     tt_cxy_pair core = {0, GetParam()};
 
     device->write_to_device(wdata.data(), wdata.size()*sizeof(uint32_t), core, 0x100, "");
-    test_utils::read_data_from_device(*device, rdata, core, 0x100, wdata.size()*sizeof(uint32_t), "");
+    device->read_from_device(rdata.data(), core, 0x100, wdata.size()*sizeof(uint32_t), "");
```

Comment:
You are reading to `rdata`, but the size of the read is expressed as `wdata.size()`. I assume these are the same size but it would be more readable if we would have `rdata.size()`.

Reply:
Yep they are the same size as these are just simple loopback tests.

Reply:
Changed to use `rdata.size()`